ostd/src/trap/handler.rs

```diff
@@ -19,7 +35,11 @@ pub(crate) fn call_irq_callback_functions(trap_frame: &TrapFrame, irq_number: us
     crate::arch::interrupts_ack(irq_number);
 
     crate::arch::irq::enable_local();
-    crate::trap::softirq::process_pending();
+
+    // Process the execution in bottom half.
+    if let Some(handler) = BOTTOM_HALF_HANDLER.get() {
+        handler()
```

Comment:
The user-given `handler` should be executed in the atomic context. Should call `disable_preempt` before entering the `handler`. The original `process_pending` includes a call to `disable_preempt`, which should be removed.